### ISSUE TITLE
(*Required) on some component labels is separated for some AT/browser combinations

### ISSUE THREAD

**Issue by bacitracin** [department-of-veterans-affairs/vets-design-system-documentation#3732]
# Bug Report

- [x]  I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

This is a tiny issue but since our CAIA person called it out, wanted to get this logged somewhere. It looks like this happens on the DateInput, Radio buttons + the Form Library labels, but not the TextInput, TextArea or Select labels.

In the Conditions team’s recent [a11y research pilot with CAIA](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97582) the accessibility specialist found that when navigating labels that include *Required, the *Required part of the label was announced separately. So she had to navigate forward an extra step. 

[Her comments:](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97582#issuecomment-2498765176)

> Star Required (*required)
> Finding:
> Star required is not connected to its according text
> Therefore, it is not announced alongside the according text.
> Recommendation: It should all announce together, to avoid the extra swipe.

<img width="606" alt="Image" src="https://github.com/user-attachments/assets/25b56a2c-81b2-4e75-966b-e53f4c0222ea" />

## What I expected to happen
When navigating the label in browse mode (in VoiceOver navigating via CTRL + OPTION + right/left arrow keys and not TAB) *Required would announce together with the label, like this screenshot:

<img width="685" alt="Image" src="https://github.com/user-attachments/assets/41f99b8d-4394-413a-aa0d-65ed7480a46c" />

## Reproducing

- CSS Library version: 48.3.0
- VoiceOver/Mac/Safari
- VoiceOver/iOS/Safari
- JAWS/Chrome

Steps to reproduce:

1. You can repro on the DateInput & the Radio Button components in Storybook
2. Using Safari turn on VoiceOver
3. Navigate through the label
4. You'll land on *Required on its own

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ]  This bug is blocking work currently in progress
- [ ]  This bug is affecting work currently in progress but we have a workaround
- [ ]  This bug is blocking work planned within the next few sprints
- [x]  This bug is not blocking any work
- [ ]  Other
